src/core_plugins/kibana/public/visualize/listing/visualize_landing_page_table.js

```diff
@@ -42,11 +74,13 @@ export class VisualizeLandingPageTable extends React.Component {
     const PAGE_SIZE = 3;
     this.props.fetch(newFilter).then((results) => {
       const items = results.hits;
+      const pagesCount = this.pager.getPagesCount(items.length);
+
       this.setState({
         items,
         isFetchingItems: false,
-        pager: new Pager(items.length, PAGE_SIZE, 1),
-        selectedIds: new SelectedIds()
+        currentPage: Math.min(this.state.currentPage, pagesCount - 1),
```

Comment:
I think we should be more explicit about whether currentPage is an index or not.  I'm working on merging your PR into mine and ran into a bug here because pagesCount can be 0.  I'm going to add a new getLastPageIndex function and put the `- 1` logic in there.  Sound okay?